### ISSUE TITLE
Filter UI: Tree view / Simple buttons

### ISSUE THREAD

**Issue by kordianbruck** [kordianbruck/miniflux#4]
Implement something along the lines of

```
All
Tag A
  |- All
  |- Feed title A
  |- Feed title B
Tag B
  |- All
  |- Feed title A
  |- Feed title C
```

If no tags are used, hide the tree fully. On mobile devices it should not be visible

**Comment by mkresin** [comment]
Just to define it more precisely. Do you agree with the following proposal which is inspired by the way [reeder2 exposes tags and feeds to the UI](https://cloud.githubusercontent.com/assets/4547948/7214565/4484e86c-e5b1-11e4-89a8-1f73416c1c5e.png)

```
All
Tag A (=All of Tag A)
  |- Feed title A
  |- Feed title B
Tag B (=All of Tag B)
  |- Feed title A
  |- Feed title C
Feed title D
Feed title E
```

**Comment by kordianbruck** [comment]
Sure, sounds good!

Dunno if the untagged items will be too many if all of them are displayed, but we can do it like that for a first draft.

**Comment by mkresin** [comment]
Well, it's up to the user to either use tagging for all feeds or don't use any tag. As soon as tags aren't used, the tree is hidden.

**Comment by kordianbruck** [comment]
Right. 

Are we actually only implementing the tree or also the simple tags on top besides the 'mark as read' links as in my suggestion back then? (See https://github.com/miniflux/miniflux/issues/251#issuecomment-91932662) 

I would suggest primarily using the buttons on top and make the tree expandable if the screen is large enough with a hover and slide out to the left. For example: in unread there should only be the tags on top, which actually contain feeds which have unread items. On the other pages it might take less sense with those buttons but rather more with the tree menu. Tree menu wont work on mobile, Tag buttons on top might just fit in.

What do you think?

**Comment by mkresin** [comment]
We should implement only one variant. Which it will be, needs to be discussed. But I am probably not very objective on that, since I prefer the treeview.

Maybe I'm a bit old fashioned, but since the days of google reader every feed reader\* has a sidebar treeview. It's hard for me to imagine, how something else should do the job.

I guess we have different requirements. I need the ability to either view unread articles of a group or if I have a lot of unread articles, only unread articles of a feed. That the way I'm working with feedly, reeder on iOS and even E-Mail. I don't see a way how this could be possible with having the tags on top.

What I find particularly disturbing with tags on top, is that space that should be used for the most important thing of feed reader - articles - is used for the tags. This could get even more problematic, if someone decides to use a lot of tags. If I have a look at my desktop screen, which has roughly double the size in horizontal than in vertical direction, I doubt that it's a good idea to add more controls to the limited vertical space.

*I had a look at feedly, digg reader, tt-rss, fever and Reeder on iOS.

**Comment by kordianbruck** [comment]
Well the only problem I see with the tree view is that it won't work for mobile. Having a 'fold in' solution for mobiles would be great. Meaning: when viewed on a small device, only the tags are displayed without structure.

**Comment by mkresin** [comment]
Yeah, we need something complete different for other screen sizes. I've no idea how the mobile view should look like, since I prefer to use native apps on my smartphone.

Miniflux already differ between mobile and desktop view by using the css classes `hide-mobile` and `hide-desktop`.

Beside the lack of ideas how the desktop or mobile view should finally look like, I've no clue how to implement it _properly_ at the moment.

**Comment by kordianbruck** [comment]
So as you actually have no better idea, I will implement my idea and we can go from there. I don't see the tree view working with miniflux design currently, so I think my solution is the next best thing.

**Comment by mkresin** [comment]
> So as you actually have no better idea, I will implement my idea and we can go from there

Well, I though I pointed out that I have an idea what could be implemented (treeview), what are my requirements and why it's a good idea to use a treeview. I've only said, that I don't know how it should look **finally**. But I refuse to say that I've no idea.

Right now, nobody else wrote a single word about requirements and expectations.

> I don't see the tree view working with miniflux design currently

Desktop view: A lot of free space on the left side. Use it for the treeview. The [bootstrap](https://github.com/miniflux/theme-bootstrap-light) and [hello](https://github.com/meradoou/hello) theme are nice examples how to use the free space.

Fell free to implement your idea. I'm sure we can fine tune stuff afterwards, like it was done with the tag selection buttons.

**Comment by kordianbruck** [comment]
![screenshot from 2015-08-02 01 46 41](https://cloud.githubusercontent.com/assets/298860/9024048/6dd384f8-38b8-11e5-96b5-bfd6736503f5.png)

I would just keep it simple. Check my latest commit which contains a basic implementation, so we can merge this to master now. We can work on the other idea later.

@mkresin @fguillot are you okey with the way it looks? Any suggestions/remarks before I open the PR?

**Comment by mkresin** [comment]
Still no beauty, but everything that I've layouted was even more ugly. So I'm fine with it, to get this finally merged.

@kordianbruck I'll test the tag stuff the next days and review the code again. I've spotted a few TODOs which shouldn't be in the final code.
